### PR TITLE
Implement SSE EventId and ReplayCursor helpers

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Implements runtime SSE helpers: a validated EventId type for SSE `id:` lines and a ReplayCursor type for handling the Last-Event-ID header, plus header extraction and error mapping. Exposes a public sse surface via crate root. Adds tests and documentation updates.

## Changes
- New files:
  - src/sse/event_id.rs
  - src/sse/replay_cursor.rs
  - src/sse/mod.rs
- Public API surface:
  - Updated src/lib.rs to re-export:
    - EventId, EventIdValidationError
    - LAST_EVENT_ID_HEADER
    - ReplayCursor, extract_replay_cursor, map_replay_cursor_error
- Documentation and planning artifacts:
  - New: docs/execplans/1-1-1-implement-sse-identifier-and-replay-cursor-helpers.md
  - New: docs/users-guide.md
  - New: docs/developers-guide.md
  - Updated: docs/roadmap.md (1.1.1 marked as done)
  - Updated: docs/contents.md (reflects new docs presence)
- Module wiring:
  - src/sse/mod.rs and crate-level re-exports to surface EventId and ReplayCursor publicly
- Tests:
  - Added unit tests for EventId and ReplayCursor (header extraction, error mapping, edge cases)
  - Tests cover missing/duplicate/empty headers, non-UTF-8 handling, and whitespace behaviour

## Rationale
 ADR 001 defines a shared SSE wire contract; this change implements the concrete, testable surface for:
 - Validated EventId for SSE `id:` lines (no CR, LF, or NULL)
 - ReplayCursor wrapping of EventId for Last-Event-ID handling
 - Public header extraction helper and error mapping to API errors
 These changes provide a stable, wire-safe API that downstream users can rely on while keeping the implementation independent of event-store specifics.

## Validation / Testing
- Unit tests for EventId validation (empty, forbidden characters, edge cases)
- Unit tests for ReplayCursor and header extraction (missing header, empty header, valid header, duplicates, non-UTF-8)
- Integration-like tests verifying AsRef/Display/From/TryFrom traits on EventId and ReplayCursor
- Error mapping tests ensuring map_replay_cursor_error yields ErrorCode::InvalidRequest with precise messages
- Formatting and lint gates pass (fmt and clippy as per repository standards)
- Documentation tests (markdownlint, nixie) are addressed by new docs files

## Documentation notes
- Public API described in docs/users-guide.md with usage examples
- Internal guidance and module layout documented in docs/developers-guide.md
- ExecPlan added under docs/execplans for task 1.1.1 (Implement validated SSE event identifier and replay cursor helpers)
- Roadmap updated to reflect completion of task 1.1.1

## Acceptance criteria alignment
- Adds runtime SSE helpers for EventId and ReplayCursor with header parsing support
- Public surface exposed via crate root
- Comprehensive unit tests for new types and header logic
- Documentation updated to reflect planned and implemented work
- Roadmap entry 1.1.1 updated to done

## Task
- See https://www.devboxer.com/task/accb7793-e8fd-4eb9-9342-623081da2035

## Summary by Sourcery

Add SSE wire-level helpers for validated event identifiers and replay cursors and expose them via the crate root, along with corresponding header parsing, error mapping, tests, and documentation updates.

New Features:
- Introduce EventId validated type and ReplayCursor wrapper for SSE event identifiers and client replay cursors.
- Add helper functions and constants for parsing the Last-Event-ID header and mapping validation errors into the shared error type.

Enhancements:
- Wire a new sse module into the crate and re-export its public API from the crate root.
- Document SSE helper usage and internal conventions in new user and developer guides, and record the execution plan for implementing SSE identifier and replay cursor helpers.

Documentation:
- Add and link a user's guide and developer's guide describing SSE helpers and crate usage patterns.
- Add an execution plan document for implementing SSE identifier and replay cursor helpers and mark the corresponding roadmap task as complete.

Tests:
- Add unit tests covering EventId validation, ReplayCursor behaviour, Last-Event-ID header extraction, error mapping, and whitespace handling.

### CODE REVIEW

File: docs/developers-guide.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+## Additional resources
+
+- [ADR 001: Shared SSE wire contract for Wildside and
+  Corbusier](adr-001-shared-sse-wire-contract-for-wildside-and-corbusier.md) —
+  normative specification for the SSE module
+- [ExecPlan: Implement SSE identifier and replay cursor
+  helpers](execplans/1-1-1-implement-sse-identifier-and-replay-cursor-helpers.md)
+   — implementation plan for task 1.1.1
+- [AGENTS.md](../AGENTS.md) — code style, testing, and commit conventions
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/developers-guide.md:255]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: src/sse/replay_cursor.rs [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+    let Some(header_value) = header_values.next() else {
+        return Ok(None);
+    };
+    if header_values.next().is_some() {
+        return Err(EventIdValidationError::InvalidHeader);
+    }
+
+    let header_text = header_value
+        .to_str()
+        .map_err(|_| EventIdValidationError::InvalidHeader)?;
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/sse/replay_cursor.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai please create a new GitHub issue for the following:

### ❌ Failed checks (1 warning)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| User-Facing Documentation | ⚠️ Warning | Pull request exports ReplayCursorError as public API without documentation in users-guide, and users-guide contains factually incorrect error type references. | Add dedicated error handling section for ReplayCursorError documenting all three variants and correct factually incorrect error type references in users-guide. |
